attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -158,31 +187,27 @@ impl Attester for TdxAttester {
     }
 
     async fn bind_init_data(&self, init_data_digest: &[u8]) -> Result<InitDataResult> {
-        let mut report = tdx_report_t { d: [0; 1024] };
-        match tdx_attest_rs::tdx_att_get_report(None, &mut report) {
-            tdx_attest_rs::tdx_attest_error_t::TDX_ATTEST_SUCCESS => {
-                log::debug!("Successfully get report")
-            }
-            error_code => {
-                bail!(
-                    "TDX Attester: Failed to get TD report. Error code: {:?}",
-                    error_code
-                );
-            }
-        };
-
-        let td_report = report
-            .d
-            .pread::<report::TdReport>(0)
-            .context("Parse TD report failed")?;
-
+        let td_report = Self::get_report()?;
         let init_data: [u8; 48] = pad(init_data_digest);
         if init_data != td_report.tdinfo.mrconfigid {
             bail!("Init data does not match!");
         }
 
         Ok(InitDataResult::Ok)
     }
+
+    async fn get_rtmr_value(&self, pcr_index: u64) -> Result<Vec<u8>> {
+        let td_report = Self::get_report()?;
+        let rtmr_index = Self::pcr_to_rtmr(pcr_index) as usize;
+
+        let mut rtmr_u8 = Vec::new();
+        let rtmr = &td_report.tdinfo.rtmr[rtmr_index * 6..(rtmr_index + 1) * 6];
+        for i in rtmr {
+            rtmr_u8.extend_from_slice(&i.to_le_bytes());
+        }
```

Comment:
it might be good to add unit tests for this part, you can use td_report fixtures and maybe move this part into an internal fn to not have the get_report() side effect.

Reply:
Nice idea. Let me add this.